pydis_site/apps/content/views/page_category.py

```diff
@@ -0,0 +1,56 @@
+import typing as t
+from pathlib import Path
+
+from django.conf import settings
+from django.http import Http404
+from django.views.generic import TemplateView
+
+from pydis_site.apps.content import utils
+
+
+class PageOrCategoryView(TemplateView):
+    """Handles pages and page categories."""
+
+    def dispatch(self, request: t.Any, *args, **kwargs) -> t.Any:
+        """Conform URL path location to the filesystem path."""
+        self.location = Path(self.kwargs.get("location", ""))
+        self.full_location = settings.PAGES_PATH / self.location
+
+        return super().dispatch(request, *args, **kwargs)
+
+    def get_template_names(self) -> t.List[str]:
+        """Checks does this use page template or listing template."""
+        if self.full_location.is_dir():
+            template_name = "content/listing.html"
+        elif self.full_location.with_suffix(".md").is_file():
+            template_name = "content/page.html"
+        else:
+            raise Http404
+
+        return [template_name]
+
+    def get_context_data(self, **kwargs) -> t.Dict[str, t.Any]:
+        """Assign proper context variables based on what resource user requests."""
+        context = super().get_context_data(**kwargs)
+
+        if self.full_location.is_dir():
+            context["categories"] = utils.get_categories(self.full_location)
+            context["category_info"] = utils.get_category(self.full_location)
+            context["content"] = utils.get_pages(self.full_location)
```

Comment:
"pages"  would be a better name than "content". At this point, "content" implies that it stores the HTML content of pages.